12-offer-encoding.md

```diff
@@ -551,144 +578,102 @@ using the `onion_message` `invoice` field.
-    - MUST reject the invoice if the `offer_id` does not refer an unexpired offer with `send_invoice`
-    - MUST reject the invoice unless the following fields are equal or unset
-      exactly as they are in the `offer`:
-      - `refund_for`
-      - `description`
-    - if `chain` is not present:
+     - MUST reject the invoice if all fields less than type 160 do not exactly match the `invoice_request`
+  - otherwise, if responding to an `offer_send_invoice` offer:
+    - MUST fail the request if `offer_send_invoice` is not present.
+    - MUST fail the request if the offer fields not do exactly match a valid, unexpired offer.
+    - if `invoice_request_chain` is not present:
        - MUST reject the invoice if bitcoin is not a supported chain for the offer.
     - otherwise:
-      - MUST reject the invoice if `chain` is not a supported chain for the offer.
-    - if the offer had a `quantity_min` or `quantity_max` field:
-      - MUST reject the invoice if there is no `quantity` field.
-      - MUST reject the invoice if there is `quantity` is not within that (inclusive) range.
+      - MUST reject the invoice if `invoice_request_chain` is not a supported chain for the offer.
+    - if `offer_quantity_min` or `offer_quantity_max` is present:
+      - MUST reject the invoice if there is no `invoice_request_quantity` field.
+      - MUST reject the invoice if `invoice_request_quantity` is not within that (inclusive) range.
     - otherwise:
-      - MUST reject the invoice if there is a `quantity` field.
-    - if the offer contained `refund_for`:
-        - MUST reject the invoice if `payer_key` does not match the invoice whose `payment_hash` is equal to `refund_for` `refunded_payment_hash`
-        - MUST reject the invoice if `refund_signature` is not set.
-        - MUST reject the invoice if `refund_signature` is not a valid signature using `payer_key` as described in [Signature Calculation](#signature-calculation).
-  - otherwise: (not an `invoice_request` reply, and no `offer_id`):
-    - if `chain` is not present:
+      - MUST reject the invoice if there is an `invoice_request_quantity` field.
+  - otherwise: (not an `invoice_request` reply, nor for `offer_send_invoice`):
+    - if `invoice_request_chain` is not present:
        - MUST reject the invoice if bitcoin is not a supported chain.
     - otherwise:
-       - MUST reject the invoice if `chain` is not a supported chain.
-  - for the bitcoin chain, if the invoice specifies `fallbacks`:
+       - MUST reject the invoice if `invoice_request_chain` is not a supported chain.
+  - for the bitcoin chain, if the invoice specifies `invoice_fallbacks`:
     - MUST ignore any `fallback_address` for which `version` is greater than 16.
     - MUST ignore any `fallback_address` for which `address` is less than 2 or greater than 40 bytes.
     - MUST ignore any `fallback_address` for which `address` does not meet known requirements for the given `version`
```

Comment:
Is there a reason for the more soft "ignore" rather than "reject" if the writer MUST write a valid witness version and program?

Reply:
Future expansion...